app/service/plugin/PluginManager.php

```diff
@@ -747,10 +711,250 @@ public function forceRegisterRoutes(string $slug): bool
+                        break;
+                    default:
+                        Route::any($routePath, $handler);
+                }
+            }
+
+            $this->routesRegistered[$slug] = true;
+        } catch (\Throwable $e) {
+            // 插件路由注册异常不影响启用
+            // 可以记录日志但不中断流程
+            try {
+                \support\Log::warning("[plugin-route-registration-failed] slug={$slug} error=" . $e->getMessage());
+            } catch (\Throwable $logError) {
+                // 忽略日志异常
+            }
+        }
+    }
+    // ========= 新增：健壮性与依赖处理 =========
+
+    /**
+     * 验证元数据与运行环境
+     * @return array{errors: array<int,string>, warnings: array<int,string>}
+     */
+    public function validate(string $slug): array
+    {
+        $errors = [];
+        $warnings = [];
+        $entry = $this->plugins[$slug] ?? null;
+        if (!$entry) {
+            $errors[] = 'Plugin not found';
+
+            return compact('errors', 'warnings');
+        }
+        $m = $entry['meta'];
+        if ($m->name === '') {
+            $warnings[] = 'Missing name';
+        }
+        if ($m->version === '') {
+            $warnings[] = 'Missing version';
+        }
+        if ($m->slug === '') {
+            $errors[] = 'Missing slug';
+        }
+        if ($m->requires_php !== '') {
+            if (!Semver::satisfies(PHP_VERSION, (string) $m->requires_php)) {
+                $errors[] = 'PHP version not satisfied: ' . PHP_VERSION . ' ! ' . $m->requires_php;
+            }
+        } elseif (($m->requires['php'] ?? '') !== '') {
+            if (!Semver::satisfies(PHP_VERSION, (string) $m->requires['php'])) {
+                $errors[] = 'PHP version not satisfied: ' . PHP_VERSION . ' ! ' . $m->requires['php'];
+            }
+        }
+        // engine 版本（可选）
+        $engine = (string) ($m->requires['engine'] ?? '');
+        if ($engine !== '') {
+            $appVer = (string) (defined('APP_VERSION') ? APP_VERSION : '1.0.0');
+            if (!Semver::satisfies($appVer, $engine)) {
+                $errors[] = 'Engine version not satisfied: ' . $appVer . ' ! ' . $engine;
+            }
+        }
+        // 冲突检查（仅已启用集）
+        $enabled = (array) (blog_config('plugins.enabled', [], true) ?: []);
+        foreach ($m->conflicts as $confSlug => $constraint) {
+            if (in_array($confSlug, $enabled, true)) {
+                $other = $this->plugins[$confSlug]['meta']->version ?? '';
+                if ($constraint === '' || $constraint === '*' || $other === '') {
+                    $errors[] = "Conflicts with {$confSlug}";
+                } elseif (Semver::satisfies($other, (string) $constraint)) {
+                    $errors[] = "Conflicts with {$confSlug} {$constraint}";
+                }
+            }
+        }
+
+        return compact('errors', 'warnings');
+    }
+
+    /**
+     * 加载已启用插件并按照依赖拓扑排序激活
+     */
+    public function loadEnabledWithDeps(): void
```

Comment:
在PluginManager中新增了依赖解析功能，这是一个重要的增强。通过DependencyResolver类实现了插件依赖的拓扑排序，确保插件按照正确的依赖顺序加载。这大大增强了插件系统的健壮性。

```suggestion
public function loadEnabledWithDeps(): void
{
    $enabled = array_values((array) (blog_config('plugins.enabled', [], true) ?: []));
    if (!$enabled) {
        return;
    }
    $metas = $this->allMetadata();
    $resolver = new DependencyResolver();
    try {
        $order = $resolver->resolve($metas, $enabled);
    } catch (PluginDependencyException $e) {
        // 回退：出现依赖错误时，按原顺序启用但跳过失败项
        $order = $enabled;
        try {
            \support\Log::warning('[plugin-deps] ' . $e->getMessage());
        } catch (\Throwable $x) {
        }
    }
    foreach ($order as $slug) {
        // 每个启用前做验证，失败则跳过
        $vr = $this->validate($slug);
        if (!empty($vr['errors'])) {
            try {
                \support\Log::warning('[plugin-validate] ' . $slug . ' ' . implode('; ', $vr['errors']));
            } catch (\Throwable $x) {
            }
            continue;
        }
        $this->enable($slug);
    }
}